Intersect.Client/Entities/Entity.cs

```diff
@@ -489,6 +489,34 @@ public void AddAnimations(List<AnimationBase> anims)
             }
         }
 
+        public virtual bool IsAllyOf(Player en)
+        {
+            if (en == null || MapInstance == default || en.MapInstance == default)
+            {
+                return false;
+            }
+
+            // Resources have no allies
+            if (Type == EntityType.Resource)
+            {
+                return false;
+            }
+
+            // Events ONLY have allies
+            if (Type == EntityType.Event)
+            {
+                return true;
+            }
+
+            // Yourself is always an ally
+            if (en.Id == Globals.Me.Id)
```

Comment:
I think this method should be comparing `en.Id` to `Id` not `Globals.Me.Id`. The `IsAllyOf()` method is being misused if the `Globals.Me.` qualification.

Reply:
Good catch, fixed